### PR TITLE
SLING-12441 Add support for binary request parameters

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/api/request/builder/Builders.java [hunk 1]
```diff
@@ -92,4 +92,16 @@ private Builders() {}
...
+
+    /**
+     * Creates a new binary request parameter
+     *
+     * @param name the parameter name
+     * @param value the parameter value
+     * @return a request parameter
+     * @since 1.3 (Sling API Bundle 2.28.0)
+     */
+    public static @NotNull RequestParameter newRequestParameter(String name, byte[] value) {
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/request/builder/Builders.java:104]: Probably allowing to parameterize the content type would be beneficial here

File: src/main/java/org/apache/sling/api/request/builder/Builders.java [hunk 2]
```diff
@@ -69,27 +69,41 @@ private Builders() {}
...
+     *
+     * @param name the parameter name
+     * @param value the parameter value
+     * @param fileName the file name (may be {@code null})
+     * @param contentType the content type (may be {@code null})
+     * @return a request parameter
+     * @since 1.3 (Sling API Bundle 2.28.0)
+     */
+    public static @NotNull RequestParameter newRequestParameter(@NotNull String name, byte @NotNull[] value, String fileName, String contentType) {
+        return new RequestParameterImpl(name, value, fileName, contentType);
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/request/builder/Builders.java:107]: @cziegeler Is there a need for binary parameters also for https://github.com/apache/sling-org-apache-sling-api/blob/2f95145f469ae175adb3d82e4e3e361236f0f2a4/src/main/java/org/apache/sling/api/request/builder/SlingHttpServletRequestBuilder.java#L37? Its `withBody(String)` method only allows Strings and does not seem to end up in the returned RequestParameterMap (https://github.com/apache/sling-org-apache-sling-api/blob/2f95145f469ae175adb3d82e4e3e361236f0f2a4/src/main/java/org/apache/sling/api/request/builder/impl/SlingHttpServletRequestImpl.java#L493)

Comment (cziegeler) [reply src/main/java/org/apache/sling/api/request/builder/Builders.java:107]: I think we can leave that out and only add if there is a real use case for it

Comment (kwin) [reply src/main/java/org/apache/sling/api/request/builder/Builders.java:107]: ok

**Comment by kwin** [comment]
If there are no further comments I will merge end of this week.

**Pull request merged**